src/ignore.c

```diff
@@ -74,6 +112,8 @@ static int does_negate_rule(int *out, git_vector *rules, git_attr_fnmatch *match
 			error = 0;
 			goto out;
 		}
+
+		error = 0;
```

Comment:
You're right, this is not required as `error` will be set in each iteration before it is read, anyway.